ntbs-service/Migrations/20210525134932_AddLegacySourceToNotification.cs

```diff
@@ -0,0 +1,24 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace ntbs_service.Migrations
+{
+    public partial class AddLegacySourceToNotification : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<string>(
+                name: "LegacySource",
+                table: "Notification",
+                type: "nvarchar(10)",
+                maxLength: 10,
+                nullable: true);
```

Comment:
I was wondering if we ought to try to default the data in the database, but I think we shouldn't because (a) we'd inevitably get some sources wrong, (b) we'd be lying about the fact that we know the source and (c) the live system doesn't have any notifications yet, so all notifications that we care about are going to be wiped at some point.

However, I'm concerned by the change in default - all existing records are now going to have the ETS ID as their legacy ID. Maybe when we calculate the legacy ID, if the source is NULL then we should fallback to the old logic? Or are you envisaging a data clean out in every environment when we deploy this code?

Reply:
I have added the fallback, seems safer.